src/board_builder/structures/pose_location.py

```diff
@@ -0,0 +1,45 @@
+import datetime
+
+import numpy as np
+from scipy.spatial.transform import Rotation as R
+
+from src.common.structures import Matrix4x4, Pose
+
+
+class PoseLocation:
+    def __init__(self, object_id):
+        self._id = object_id
+        self._timestamp = str(datetime.datetime.now())
+
+        self.__TMatrix = np.eye(4)
+        self.__RMAT_list = []  # Rotation matrix
+        self.__TVEC_list = []  # Translation vector
+
+        self.frame_count = 0
+
+    def add_matrix(self, transformation_matrix, timestamp):
+        self._timestamp = timestamp
+
+        self.__RMAT_list.append(transformation_matrix[:3, :3])
+        self.__TVEC_list.append(transformation_matrix[:3, 3])
+
+        avg_translation = np.mean(self.__TVEC_list, axis=0)
+
+        quaternions = [R.from_matrix(rot).as_quat() for rot in self.__RMAT_list]
+        avg_quat = np.mean(quaternions, axis=0)
+        avg_quat = avg_quat / np.linalg.norm(avg_quat)  # Normalize the average quaternion
```

Comment:
I think the idea to average a quaternion is appropriate, but the approach here might not be mathematically correct. Take a look at `pose_solver.util.average_quaternion.py` and look at the links/references at the top of that file. You can move that function to `common.util.average_quaternion`.